drivers/sensor/sensor_shell.c

```diff
@@ -158,6 +233,115 @@ static int cmd_get_sensor(const struct shell *shell, size_t argc, char *argv[])
+		struct sensor_value value = {0};
+
+		if (channel < 0) {
+			shell_error(shell_ptr, "Channel '%s' unknown", argv[i]);
+			return -EINVAL;
+		}
+		if (attr < 0) {
+			shell_error(shell_ptr, "Attribute '%s' unknown", argv[i + 1]);
+			return -EINVAL;
+		}
+		if (parse_sensor_value(argv[i + 2], &value)) {
+			shell_error(shell_ptr, "Sensor value '%s' invalid", argv[i + 2]);
+			return -EINVAL;
+		}
+
+		rc = sensor_attr_set(dev, channel, attr, &value);
+		if (rc) {
+			shell_error(shell_ptr, "Failed to set channel(%s) attribute(%s): %d",
+				    sensor_channel_name[channel], sensor_attribute_name[attr], rc);
+			continue;
+		}
+		shell_info(shell_ptr, "%s channel=%s, attr=%s set to value=%s", dev->name,
+			   sensor_channel_name[channel], sensor_attribute_name[attr], argv[i + 2]);
+	}
+	return 0;
+}
+
+static void cmd_sensor_attr_get_handler(const struct shell *shell_ptr, const struct device *dev,
+					const char *channel_name, const char *attr_name,
+					bool print_missing_attribute)
+{
+	int channel =
+		parse_named_int(channel_name, sensor_channel_name, ARRAY_SIZE(sensor_channel_name));
+	int attr = parse_named_int(attr_name, sensor_attribute_name,
+				   ARRAY_SIZE(sensor_attribute_name));
+	struct sensor_value value = {0};
+	int rc;
+
+	if (channel < 0) {
+		shell_error(shell_ptr, "Channel '%s' unknown", channel_name);
+		return;
+	}
+	if (attr < 0) {
+		shell_error(shell_ptr, "Attribute '%s' unknown", attr_name);
+		return;
+	}
+
+	rc = sensor_attr_get(dev, channel, attr, &value);
+
+	if (rc != 0) {
+		if (rc == -EINVAL && !print_missing_attribute) {
+			return;
+		}
+		shell_error(shell_ptr, "Failed to get channel(%s) attribute(%s): %d",
+			    sensor_channel_name[channel], sensor_attribute_name[attr], rc);
+		return;
+	}
+
+	shell_info(shell_ptr, "%s(channel=%s, attr=%s) value=%.6f", dev->name,
+		   sensor_channel_name[channel], sensor_attribute_name[attr],
+		   sensor_value_to_double(&value));
+}
+
+static int cmd_sensor_attr_get(const struct shell *shell_ptr, size_t argc, char *argv[])
+{
+	const struct device *dev;
+
+	dev = device_get_binding(argv[1]);
+	if (dev == NULL) {
+		shell_error(shell_ptr, "Device unknown (%s)", argv[1]);
+		return -ENODEV;
+	}
+
+	if (argc > 2) {
+		for (size_t i = 2; i < argc; i += 2) {
+			cmd_sensor_attr_get_handler(shell_ptr, dev, argv[i], argv[i + 1],
+						    /*print_missing_attribute=*/true);
+		}
+	} else {
+		for (size_t channel_idx = 0; channel_idx < ARRAY_SIZE(sensor_channel_name);
+		     ++channel_idx) {
+			for (size_t attr_idx = 0; attr_idx < ARRAY_SIZE(sensor_attribute_name);
+			     ++attr_idx) {
+				cmd_sensor_attr_get_handler(shell_ptr, dev,
+							    sensor_channel_name[channel_idx],
+							    sensor_attribute_name[attr_idx],
+							    /*print_missing_attribute=*/false);
+			}
+		}
+	}
+	return 0;
+}
+
 static void channel_name_get(size_t idx, struct shell_static_entry *entry);
 
 SHELL_DYNAMIC_CMD_CREATE(dsub_channel_name, channel_name_get);
```

Comment:
I wonder if it's possible to get tab completion for the attribute name?  You could try:
```
static void channel_name_get(size_t idx, struct shell_static_entry *entry)
{
	/* ... */
	if (get_attr || set_attr) {
		/* Process attribute */
		entry->subcmd = &dsub_attribute_name;
	} else {
		/* Continue processing channels */
		entry->subcmd = &dsub_channel_name;
	}
	/* ... */
}
```
Your `attribute_name_get()` function would set `entry->subcmd` to NULL to terminate the processing.

Reply:
Done